cmd/notify-mailer/main.go

```diff
@@ -89,36 +96,36 @@ func (m *mailer) printStatus(to string, cur, total int, start time.Time) {
 		to, cur, total, completion, elapsed)
 }
 
-// uniq returns a slice of strings consisting of the input slice with all
-// duplicates removed. It preserves the ordering of the input slice.
-func uniq(input []string) []string {
-	var output []string
-	uniqMap := map[string]bool{}
-	for _, s := range input {
-		// Only append to the output items that have not been seen.
-		if _, ok := uniqMap[s]; !ok {
-			output = append(output, s)
-		}
-		uniqMap[s] = true
+func sortAddresses(input emailToRecipientMap) []string {
+	var addresses []string
+	for k, _ := range input {
+		addresses = append(addresses, k)
 	}
-	return output
+	sort.Strings(addresses)
+	return addresses
 }
 
 func (m *mailer) run() error {
 	if err := m.ok(); err != nil {
 		return err
 	}
 
-	destinations, err := m.resolveDestinations()
+	m.log.Infof("Resolving %d destination addresses", len(m.destinations))
+	addressesToRecipients, err := m.resolveEmailAddresses()
```

Comment:
Can you add a check for `len(addressesToRecipients) == 0`? I think based on using `range` elsewhere it won't be super obvious where this condition will first fail and catching it here seems sensible.